### PR TITLE
Clarify Catalog Layout Best Practices for collections

### PR DESCRIPTION
**Pull request by CloudNiner**
**Related Issue(s):** #925

**Proposed Changes:**

1. This PR is an attempt to clarify best practices catalog and collection naming conventions in the catalog layout section

In practice, we've found that every Catalog is named catalog.json
and every Collection is named collection.json, which is enforced
by tools such as PySTAC. The change to suggest that Collections at
the root of a STAC Catalog should be called collection.json is
discussed in #925 

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] ~~I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or**~~ a CHANGELOG entry is not required.
- [ ] ~~This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.~~

### CODE REVIEW

File: best-practices.md [hunk 1]
```diff
@@ -205,8 +205,8 @@ We encourage people to explore new structures of linking data, but the following
 ended up doing. Following these recommendations makes for more legible catalogs.
 
 1. Root documents (catalogs / collections) should be at the root of a directory tree containing the static catalog.
-2. Catalogs should be named `catalog.json` (cf. `index.html`).
-3. Collections that are distinct from catalogs should be named `collection.json`.
+2. Catalogs should be named `catalog.json` and Collections should be named `collection.json`.
+3. Collections at the root of a Catalog should be named `collection.json`.
```
[the remark this answers is not in the document]
Comment (CloudNiner) [reply best-practices.md:209]: Hmm I suppose it does, was just trying to make explicit the discussion over in #925. Maybe there's a better way to say this?

[the remark this answers is not in the document]
Comment (m-mohr) [reply best-practices.md:207]: Okay, I guess we could remove it as 2 is pretty clear, but I'm also fine to leave it in.

**Review by m-mohr — APPROVED** [review]

File: best-practices.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cholmes) [reply best-practices.md:207]: Yeah, I think we should remove rule 3, as it follows from 1 & 2, and I could see people trying to read more into it. I think it's clear with just the two rules. @CloudNiner - could you remove that line, and then we can merge? And thanks for the contribution!

[the remark this answers is not in the document]
Comment (CloudNiner) [reply best-practices.md]: 👍 updated!

**Review by cholmes — APPROVED** [review]

**Pull request merged**